torch/csrc/distributed/c10d/init.cpp

```diff
@@ -186,6 +186,9 @@ PyObject* c10d_init(PyObject* _unused) {
 An enum-like class for available reduction operations: ``SUM``, ``PRODUCT``,
 ``MIN``, ``MAX``, ``BAND``, ``BOR``, and ``BXOR``.
 
+Note that ``BAND``, ``BOR``, and ``BXOR`` reductions are not available when
+using the ``NCCL`` backend.
+
```

Comment:
Do we return an appropriate exception when these ops are used with NCCL? If so, can we add tests that validate this?

Reply:
Yes, this was done as part of https://github.com/pytorch/pytorch/pull/42669/ and there are tests for the exception.